command/agent/fs_endpoint_test.go

```diff
@@ -342,6 +342,45 @@ func TestHTTP_FS_Cat(t *testing.T) {
 }
 
 func TestHTTP_FS_Stream(t *testing.T) {
+	t.Parallel()
+	require := require.New(t)
+	httpTest(t, nil, func(s *TestAgent) {
+		a := mockFSAlloc(s.client.NodeID(), nil)
+		addAllocToClient(s, a, terminalClientAlloc)
+
+		offset := 4
+		expectation := base64.StdEncoding.EncodeToString(
+			[]byte(defaultLoggerMockDriverStdout[len(defaultLoggerMockDriverStdout)-offset:]))
+		path := fmt.Sprintf("/v1/client/fs/stream/%s?path=alloc/logs/web.stdout.0&offset=%d&origin=end&follow=false",
+			a.ID, offset)
+
+		p, _ := io.Pipe()
+		req, err := http.NewRequest("GET", path, p)
+		require.Nil(err)
+		respW := testutil.NewResponseRecorder()
+		go func() {
+			_, err = s.Server.Stream(respW, req)
+			require.Nil(err)
```

Comment:
I think we need to borrow the `close(doneCh)` here as well but we need to check later that it does close, so reverse the logic in other follow=true case: https://github.com/hashicorp/nomad/blob/293d2123c2ae448329954f52747c64a65a994516/command/agent/fs_endpoint_test.go#L504-L508 .

FWIW, this and the existing test aren't proper in using `require.` assertions in a goroutine.  They don't behave well, but don't worry about it now :).

Reply:
> FWIW, this and the existing test aren't proper in using require. assertions in a goroutine. They don't behave well, but don't worry about it now :).

Ok I'll leave this for now but yeah that sounds like it'd cause some interesting behavior on failure. 😁